File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    [...s].forEach((value1, index1) => {
        const arr = [...s].slice(0, index1).reverse();
        const flag = arr.some((value2, index2) => {
            if(value2 === value1) {
                answer.push(index2+1);
                return true;
            }
            return false;
        })
        if(!flag){
            answer.push(-1);
        }
        
    })
    return answer;
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    last_index = {}
    answer = []
    
    for idx, ch in enumerate(s):
        if ch in last_index:
            answer.append(idx - last_index[ch])
        else:
            answer.append(-1)
        last_index[ch] = idx
    
    return answer